api/v1/controllers/AuthController.js

```diff
@@ -70,8 +79,58 @@ function refreshToken (req, res, next) {
 		});
 }
 
+function requestPasswordReset (req, res, next) {
+	var userEmail = req.query.email;
+
+	UserService
+		.findUserByEmail(userEmail)
+		.then(function (user){
+			return TokenService.generateToken(user, scopes.AUTH);
+		})
+		.then(function (tokenVal){
+			//TODO: Determine exact url route for reset page
+			tokenURL = config.uri.passwordReset + tokenVal;
+			var substitutions = {'resetURL': tokenURL};
+			return MailService.send(userEmail, mail.templates.passwordReset, substitutions);
+		})
+		.then(function(){
+			res.body = {};
+			next();
+			return null;
+		})
+		.catch(function (error){
+			next(error);
+			return null;
+		});
+}
+
+function passwordReset(req, res, next) {
+	TokenService
+		.findTokenByValue(req.body.token, 'AUTH')
+		.then(function (token) {
+			token.destroy();
+			return UserService.resetPassword(token.related('user'), req.body.password);
+		})
+		.then(function (user) {
+			return AuthService.issueForUser(user);
+		})
+		.then(function (auth) {
+			res.body = {};
+			res.body.auth = auth;
+			next();
+			return null;
+		})
+		.catch(function (error) {
+			next(error);
+			return null;
+		});
+};
+
+
 router.post('', createToken);
 router.get('/refresh', refreshToken);
+router.get('/reset', requestPasswordReset);
```

Comment:
Not a huge deal, but why use a GET request and a query string containing the email for requesting a password reset instead of a POST with the email in the body?